arangod/Aql/OptimizerRules.cpp

```diff
@@ -3701,15 +3700,70 @@ void arangodb::aql::scatterInClusterRule(Optimizer* opt, std::unique_ptr<Executi
         deps[0]->getFirstDependency()->getType() == ExecutionNode::DISTRIBUTE) {
       continue;
     }
+    
+    auto const nodeType = node->getType();
+    bool scatterOnly = false;
 
+    // check if the smart-joins rule marked this node as not requiring a full scatter..gather setup.
+    // in case of local joins we want to avoid inserting scatter..gather of course.
     if (plan->shouldExcludeFromScatterGather(node)) {
```

Comment:
Just so I understand correctly what's happening here: In case of a smart join, this relies upon the fact that the "upper" node of the two is the excluded one. So the non-excluded one gets Scatter/Gather around it first. Now for this excluded node, either only the Scatter is inserted - in case any coordinator snippet is above it - or nothing at all.
And if the Scatter is inserted, Remote/Scatter below is removed.
On the other hand, if nothing is inserted, it still relies on the `removeUnnecessaryRemoteScatter` rule to remove it.

Wouldn't it be good to *always* remove the `Remote/Scatter` here, and just remove the dependence upon the other rule completely? Seems little work at first glance and would make all this at least a little simpler and less interdependent.